src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -5,20 +5,107 @@
  * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private Entry<K, V>[] table;
+    private int size;
+    private int capacity = (int) (INITIAL_CAPACITY * LOAD_FACTOR);
+
+    private static class Entry<K, V> {
+        private K key;
+        private V value;
+        Entry<K, V> next;
+
+        private Entry(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
+
+    public MyHashMap() {
+        this.table = new Entry[INITIAL_CAPACITY];
+    }
+
+    private int hashCode(K key) {
+        if (key == null) {
+            return 0;
+        } else {
+            return Math.abs(key.hashCode()) % table.length;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
+        int index = hashCode(key);
+        if (table[index] == null) {
+            table[index] = new Entry<>(key, value);
+            size++;
+            return;
+        }
+        if (index == 0) {
```

Comment:
what does this if statement do?

Reply:
variable index calculated hashcode from key. when key=null in method hashCode index=0. so this statement get true result in putAndGetByNullKey test